zrokdir/client.go

```diff
@@ -27,7 +27,10 @@ func (zrd *ZrokDir) Client() (*rest_client_zrok.Zrok, error) {
 	if err != nil {
 		return nil, errors.Wrapf(err, "error getting version from api endpoint '%v': %v", apiEndpoint, err)
 	}
-	if !strings.HasPrefix(string(v.Payload), build.Series) {
+	// allow reported version string to be optionally prefixed with
+	// "refs/heads/" or "refs/tags/"
+	re := regexp.MustCompile(`^(refs/(heads|tags)/)?` + build.Series)
+	if ! re.MatchString(string(v.Payload)) {
```

Comment:
it looks like this breaks release functionality (where the version will start with `v...`). will handle this another way.

Reply:
It shouldn't break that, no. The `refs/tags/` and `refs/heads/` prefixes are tolerated and optional in this implementation (see trailing `?` after `(heads|tags)`, not required. That is, for a `build.Series` value of `v0.4`, valid `v.Payload` globs include `refs/heads/v4.0*` and `v0.4*`.

Reply:
so our release versions will always be prefixed `refs/tags/v0.4`? that version information is used to validate that you're talking to the correct api version for a client. that needs to work for release builds, and i'm not sure that's happening with your change.